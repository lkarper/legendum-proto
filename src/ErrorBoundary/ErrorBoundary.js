import React, { Component } from 'react';

class ErrorBoundary extends Component {
    state = {
        hasError: false,
    }

    static getDerivedStateFromError(error) {
        return { hasError: true };
    }

    render() {
        if (this.state.hasError) {
            return (
                <section className='section'>
                    <header>
                        <h2>Error</h2>
                    </header>
                    <p>Looks like something went wrong.  Check the address, check your connection, and try again.</p>
                </section>
            );
        }
        return this.props.children;
    }
}

export default ErrorBoundary;
