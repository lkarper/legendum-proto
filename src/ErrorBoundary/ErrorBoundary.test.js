import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ErrorBoundary from './ErrorBoundary';

const Test = () => null;

describe('Error boundary', () => {
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <ErrorBoundary>
                <p>Test</p>
            </ErrorBoundary>,
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    });

    it('catches errors in children', () => {
        const wrapper = mount(
            <ErrorBoundary>
                <Test />
            </ErrorBoundary>
        );
        const error = new Error('test');
        wrapper.find(Test).simulateError(error);
        expect(toJSON(wrapper)).toMatchSnapshot();
    });
});
