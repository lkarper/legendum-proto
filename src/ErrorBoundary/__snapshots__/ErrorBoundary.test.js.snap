// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error boundary catches errors in children 1`] = `
<ErrorBoundary>
  <section
    className="section"
  >
    <header>
      <h2>
        Error
      </h2>
    </header>
    <p>
      Looks like something went wrong.  Check the address, check your connection, and try again.
    </p>
  </section>
</ErrorBoundary>
`;
